src/hooks/useProfiles.ts

```diff
@@ -105,5 +83,20 @@ export const useProfiles = (
   if (!canManageProfiles && query.data?.responseData.canManageProfiles) {
     useAccountStore.setState({ canManageProfiles: true });
   }
-  return { ...query, profilesEnabled: query.data?.responseData.canManageProfiles && canManageProfiles };
+
+  return {
```

Comment:
If this is like this just for the avatar, maybe it can be split somehow..I don't see the issue now, it's just a little hard to read, but in the future if we need to handle more properties here, it may be problematic.

Reply:
The same can be achieved a bit more clean. It's not a good idea to spread the `query` object because query-query will track all observed properties and trigger a rerender when these tracked properties update. I don't know if this is enabled by default in this version, but it is in the latest.

I would also use the `onSuccess` and `select` properties to set the `canManageProfiles` state once (without checking this every render) and restructure the data.

PS Can we prevent changing the store from this query? It is a weird side-effect, IMHO...

```tsx
  return useQuery(['listProfiles'], listProfiles, { 
    ...options, 
    enabled: !!user,
    onSuccess: (data) => {
      // update the canManageProfiles in state based on this response
      if (!useAccountStore.getState().canManageProfiles && data.responseData.canManageProfiles) {
        useAccountStore.setState({ canManageProfiles: true });
      }
    },
    select: (data) => ({
      ...data,
      responseData: {
        ...data.responseData,
        collection: data.responseData.collection.map((profile) => ({
          ...profile,
          avatar_url: profile?.avatar_url || defaultAvatar,
        })),
      },
    }),
  });
```

Even better would be to move the default avatar_url to the service call because then it will be cached by react-query, and we don't need to create a new object every render.